### PR TITLE
Chronos: reset batch_size according to `forecaster.num_processes`

### PR DESCRIPTION
**Pull request by Rebekk**
related to #4475

### CODE REVIEW

File: python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py [hunk 1]
```diff
@@ -140,7 +140,7 @@ def fit(self, data, epochs=1, batch_size=32):
             if isinstance(data, tuple):
                 data = DataLoader(TensorDataset(torch.from_numpy(data[0]),
                                                 torch.from_numpy(data[1])),
-                                  batch_size=batch_size,
+                                  batch_size=batch_size//self.num_processes,
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py:143]: We should report a warning if `batch_size` can not be divided with no remainder by `self.num_processes`

Comment (Rebekk) [reply python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py:143]: OK

File: python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py [hunk 2]
```diff
@@ -138,9 +138,12 @@ def fit(self, data, epochs=1, batch_size=32):
 
             # data transformation
             if isinstance(data, tuple):
+                if batch_size % self.num_processes != 0:
+                    warnings.warn("'batch_size' cannot be divided with no remainder by "
+                                  "'self.num_processes'.")
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py:143]: maybe we can should them the exact value of batch_size and self.num_processes

Comment (TheaperDeng) [reply python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py:145]: oh basically, we only want the warning message contains the rich output.
warnings.warn(f'......, we got batch_size = {batch_size} and "num of processes" = {self.num_processes}')

**Comment by Rebekk** [comment]
Jenkins:
http://10.112.231.51:18888/job/BigDL-Chronos-PR-Validation/776/
http://10.112.231.51:18888/job/BigDL-PRVN-chronos-Python-Spark-2.4-py36-ray-part1/823/
http://10.112.231.51:18889/view/BigDL-PR-Validation/job/BigDL-Chronos-PR-Validation/381/

**Comment by TheaperDeng** [comment]
http://10.112.231.51:18888/job/BigDL-Chronos-PR-Validation/895/
http://10.112.231.51:18888/job/BigDL-PRVN-chronos-Python-Spark-3.1-py37-Horovod/1010/

File: python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py (hunk 1 above)
Comment (TheaperDeng) [reply python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py:148]: just realized this should be `batch_size=max(1, batch_size//self.num_processes)`

**Comment by TheaperDeng** [comment]
http://10.112.231.51:18888/job/BigDL-Chronos-PR-Validation/899/
http://10.112.231.51:18888/job/BigDL-PRNV-chronos-Python-Spark-2.4-py37-Horovod/1016/

**Pull request merged**